front-end/src/app/layout/committee-banner/committee-banner.component.ts

```diff
@@ -6,24 +6,67 @@ import { Subject } from 'rxjs';
 @Component({
   selector: 'app-committee-banner',
   templateUrl: './committee-banner.component.html',
-  styleUrls: ['./committee-banner.component.scss']
+  styleUrls: ['./committee-banner.component.scss'],
 })
 export class CommitteeBannerComponent implements OnInit, OnDestroy {
-  committeeName: string | undefined;
-  subBannerItems: (string | undefined)[] = [];
+  committeeName?: string;
+  committeeStatus?: string;
+  committeeFrequency?: string;
+  committeeType?: string;
+  committeeID?: string;
   private destroy$ = new Subject<boolean>();
 
-  constructor(private store: Store) { }
+  constructor(private store: Store) {}
+
+  public committee_statuses = {
+    T: 'Terminated',
+    A: 'Administratively terminated',
+    D: 'Active - Debt',
+    W: 'Active - Waived',
+    M: 'Active - Monthly filer',
+    Q: 'Active - Quarterly filer',
+  };
+
+  public committee_types = {
+    C: 'communication cost',
+    D: 'delegate',
+    E: 'electioneering communication',
+    H: 'House',
+    I: 'independent expenditure filer (not a committee)',
+    N: 'PAC - nonqualified',
+    O: 'independent expenditure-only (super PACs)',
+    P: 'presidential',
+    Q: 'PAC - qualified',
+    S: 'Senate',
+    U: 'single candidate independent expenditure',
+    V: 'PAC with non-contribution account, nonqualified',
+    W: 'PAC with non-contribution account, qualified',
+    X: 'party, nonqualified',
+    Y: 'party, qualified',
+    Z: 'national party non-federal account',
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private getArbitraryValueFromObject(code: string, object: any): string {
+    return object[code as keyof typeof object] || '';
+  }
```

Comment:
Typescript was not happy with `committee_types[frequency_code] || ''` and this was the _least_ obnoxious work-around that I could get going.

Reply:
ah you need a type for the map

try this:
```
public committee_statuses: { [key: string]: string } = {
```
and then just use:
```
this.committeeFrequency = this.committee_statuses[frequency_code] || '';
```

Reply:
:woman_facepalming: